src/conditions/check_fields.rs

```diff
@@ -236,12 +250,29 @@ impl ConditionConfig for CheckFieldsConfig {
 //------------------------------------------------------------------------------
 
 pub struct CheckFields {
-    predicates: Vec<Box<dyn CheckFieldsPredicate>>,
+    predicates: IndexMap<String, Box<dyn CheckFieldsPredicate>>,
 }
 
 impl Condition for CheckFields {
     fn check(&self, e: &Event) -> bool {
-        self.predicates.iter().find(|p| !p.check(e)).is_none()
+        self.predicates.iter().find(|(_, p)| !p.check(e)).is_none()
+    }
+
+    fn check_with_context(&self, e: &Event) -> Result<(), String> {
+        let failed_preds = self
+            .predicates
+            .iter()
+            .filter(|(_, p)| !p.check(e))
+            .map(|(n, _)| n.to_owned())
+            .collect::<Vec<_>>();
+        if failed_preds.is_empty() {
+            Ok(())
+        } else {
+            Err(format!(
+                "predicates failed: [ {} ]",
```

Comment:
Hmm, is there a reason you reach for this formatting over `Debug`?

Reply:
Just avoiding the unnecessary quotes, so we get `predicates failed: [ foo.eq: 'bar' ]` instead of `predicates failed: [ "foo.eq: 'bar'" ]`.